src/modules/launcher/Plugins/Microsoft.PowerToys.Run.Plugin.Calculator.UnitTest/NumberTranslatorTests.cs

```diff
@@ -132,12 +132,26 @@ public void Translate_RemoveNumberGroupSeparator_WhenCalled(string decimalSepara
         }
 
         [DataTestMethod]
-        [DataRow("12,0004", "12.0004")]
-        [DataRow("0xF000", "0xF000")]
-        public void Translate_NoRemovalOfLeadingZeroesOnEdgeCases(string input, string expectedResult)
+        [DataRow("de-DE", "12,0004", "12.0004")]
+        [DataRow("de-DE", "0xF000", "0xF000")]
+        [DataRow("de-DE", "0", "0")]
+        [DataRow("de-DE", "00", "0")]
+        [DataRow("de-DE", "12.0004", "120004")] // . is the group separator in de-DE
+        [DataRow("de-DE", "12.04", "1204")]
+        [DataRow("de-DE", "12.4", "124")]
+        [DataRow("de-DE", "123.01 + 52.30", "12301 + 5230")]
+        [DataRow("de-DE", "123.001 + 52.30", "123001 + 5230")]
+        [DataRow("fr-FR", "0", "0")]
+        [DataRow("fr-FR", "00", "0")]
+        [DataRow("fr-FR", "12.0004", "12.0004")] // . is not decimal or group separator in fr-FR
```

Comment:
Also a group of 4

Reply:
yeah, just wanted to make sure we were keeping all 0s I don't think there's an issue with showing them in groups of 4, since when people are inputting this, they usually mean decimal separator. (although if it's used as a group separator in a language, we try to treat it as such)